File: MacTemplet/NNCagtegoryPro/NSSegmentedControl+Helper.m
//
//  NSSegmentedControl+Helper.m
//  MacTemplet
//
//  Created by Bin Shang on 2019/6/10.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

#import "NSSegmentedControl+Helper.h"
#import <objc/runtime.h>

@implementation NSSegmentedControl (Helper)

-(NSArray<NSString *> *)items{
    return objc_getAssociatedObject(self, _cmd);
}

-(void)setItems:(NSArray<NSString *> *)items{
    objc_setAssociatedObject(self, @selector(items), items, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
    self.segmentCount = items.count;
    self.selectedSegment = 0;
    for (NSInteger i = 0; i < items.count; i++) {
        [self setLabel:items[i] forSegment:i];
//        NSLog(@"%@_%@_%@",NSStringFromSelector(_cmd), self, items[i]);
    }
}



@end

File: MacTemplet/MacCagtegory/NSSegmentedControl+Helper.swift
//
//  NSSegmentedControl+Helper.swift
//  MacTemplet
//
//  Created by Bin Shang on 2019/11/23.
//  Copyright © 2019 Bin Shang. All rights reserved.
//

import Cocoa

@objc public extension NSSegmentedControl {

    var items: [String] {
        get {
            return objc_getAssociatedObject(self, RuntimeKeySelector(#function)) as! [String]
        }
        set {
            objc_setAssociatedObject(self, RuntimeKeySelector(#function), newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC);
            
            segmentCount = newValue.count;
            selectedSegment = 0;
            for e in newValue.enumerated() {
                self.setLabel(e.element, forSegment: e.offset)
            }
        }
    }
    
}
